### ISSUE TITLE
The browser launched by cypress can't access the internet

### ISSUE THREAD

**Issue by ghostinthemuf** [cypress-io/cypress#27625]
### Current behavior

When running my test suite, the browser launched by cypress can't seem to access any website whatsoever, running in open mode or headless mode.

For example, in headless mode I get this :

```
ERR_EMPTY_RESPONSE (-324) loading 'http://0.0.0.0:18081/__/#/tests/integration/webBuybackWeb2Store.feature'
Error: ERR_EMPTY_RESPONSE (-324) loading 'http://0.0.0.0:18081/__/#/tests/integration/webBuybackWeb2Store.feature'
    at rejectAndCleanup (electron/js2c/browser_init.js:161:7486)
    at Object.failListener (electron/js2c/browser_init.js:161:7699)
    at Object.emit (events.js:376:20)
```
Same test in open mode, I get this : 

![image](https://github.com/cypress-io/cypress/assets/51634162/706cb8d5-9788-4842-b296-31e2b32a900b)

Of course, my application is indeed up and running at http://0.0.0.0:18081/, I can access it with no problem in my "own" chrome instance.

And anyway, in open mode I can see that the browser can't access any domain : 

![image](https://github.com/cypress-io/cypress/assets/51634162/e1d6117e-7628-4bb9-bb7c-554466173975)

### Desired behavior

Access my application running at http://0.0.0.0:18081/login, or any other domain for that matter

### Test code to reproduce

Not sure it's possible. Anyway my issue is probably caused by some local configuration (i'm part of a team and other members on the same repo don't have my problem).

### Cypress Version

Cypress package version: 8.7.0 Cypress binary version: 8.7.0 Electron version: 14.1.0 Bundled Node version: 14.17.0

### Node version

v16.20.1

### Operating System

Ubuntu 22.04.3 LTS

### Other

I'm not using a VPN.
Unfortunately, upgrading cypress is not an option at the moment in the project i'm working.

**Comment by MikeMcC399** [comment]
@ghostinthemuf 

`http://0.0.0.0` is not typically used for this purpose.

Can you use `http://localhost` or `http://127.0.0.1` instead?

**Comment by ghostinthemuf** [comment]
Hello @MikeMcC399, thanks for reaching out !

You mean changing the way the application is served ? I can try that (note that it's running in a docker container), but as I said, the app is indeed up, and accessible via another browser. And meanwhile, the cypress browser can't access any domain. That's why I'm suspecting a configuration problem, but I couldn't find any relatable issue online :/

**Comment by ghostinthemuf** [comment]
When using Firefox instead of Chrome (in open mode), this is what I get : 

**Title:** Error launching browser

**Message:** Cypress failed to make a connection to Firefox.

This usually indicates there was a problem opening the Firefox browser.

Error details:

Error: Cypress failed to make a connection to the Chrome DevTools Protocol after retrying for 50 seconds.

This usually indicates there was a problem opening the Firefox browser.

The CDP port requested was <span class="ansi-yellow-fg">45675</span>.

Error details:

Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)

    at get (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/errors.js:1035:15)
    at Object.throwErr [as throw] (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/errors.js:1053:9)
    at /home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/browsers/protocol.js:113:21
    at tryCatcher (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:725:18)
    at _drainQueueStep (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:93:12)
    at _drainQueue (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:86:9)
    at Async._drainQueues (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:102:5)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:15:14)
    at processImmediate (internal/timers.js:462:21)


**Stack trace:**
```
Error: Cypress failed to make a connection to Firefox.

This usually indicates there was a problem opening the Firefox browser.

Error details:

Error: Cypress failed to make a connection to the Chrome DevTools Protocol after retrying for 50 seconds.

This usually indicates there was a problem opening the Firefox browser.

The CDP port requested was [33m45675[39m.

Error details:

Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)

    at get (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/errors.js:1035:15)
    at Object.throwErr [as throw] (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/errors.js:1053:9)
    at /home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/browsers/protocol.js:113:21
    at tryCatcher (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/promise.js:725:18)
    at _drainQueueStep (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:93:12)
    at _drainQueue (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:86:9)
    at Async._drainQueues (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:102:5)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/bluebird/js/release/async.js:15:14)
    at processImmediate (internal/timers.js:462:21)

    at get (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/errors.js:1035:15)
    at Object.throwErr [as throw] (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/errors.js:1053:9)
    at Object.<anonymous> (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/lib/browsers/firefox.js:422:25)
    at Generator.throw (<anonymous>:null:null)
    at rejected (/home/antoine/.cache/Cypress/8.7.0/Cypress/resources/app/packages/server/node_modules/tslib/tslib.js:115:69)

```

**Comment by MikeMcC399** [comment]
@ghostinthemuf 

I missed the point that others on your team do not have your problem, so that does suggest a configuration problem on your system. Perhaps you should be comparing your system to the setup on your colleagues systems?

---
You could try using a fully-functional demo system from Cypress:

```bash
git clone https://github.com/cypress-io/cypress-example-kitchensink.git
cd cypress-example-kitchensink
npm install
npm run local:open
```

select "E2E Testing" when Cypress opens.

That would give you an independent check.

---

This issue list is for bugs and feature requests and it is not intended for configuration issues. You can get community help from the Cypress technical community on Discord

[![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord)
(click on button)

The `8.7.0` version of Cypress you are using is a legacy version which has the Node.js `14` version built in, which already entered end-of-life.

**Comment by MikeMcC399** [comment]
@ghostinthemuf 

Your additional comment about Firefox makes this issue more complex. I suggest that you concentrate on getting your system to work with Electron or Chrome first.

If you are using a Cypress Docker image with Firefox then you need to specify a user. See https://github.com/cypress-io/cypress-docker-images#firefox-not-found.

**Comment by ghostinthemuf** [comment]
Thanks for your leads. Cypress itself is running locally on my machine, and the app I'm testing is in docker container.

But anyway, I tried running the demo system as you suggested, and the same problem occurs (cypress browser can't connect to anything) ! So it's not tied to my project, nor to the version of cypress...

This is so weird :face_with_spiral_eyes:

**Comment by jennifer-shehane** [comment]
Yeah, I haven’t seen this issue with Cypress before and agree it’s likely due to some local configuration. I hope you figure it out.

**Comment by unirora** [comment]
@ghostinthemuf  Did you able to fix this issue? if yes, how?

**Comment by ghostinthemuf** [comment]
@unirora sorry I don't have a very satisfactory answer. I suspected that the problem lied in my machine (maybe another program preventing Cypress from accessing the internet connection, using the same ports, whatever), so I uninstall a bunch of stuff that came preinstalled on my Ubuntu distro... and it worked. Cypress internet connection was restored. But I didn't do it step by step, so I couldn't pinpoint the culprit. Anyway, this was probably a very specific scenario, so you should probably create a new issue with your own specs. Hope this helps a little bit !

**Comment by Simonmie** [comment]
I also encountered the issue of Cypress not being able to connect to the internet when I first started using it. On my computer, I seemed to have found a solution. I learned that Cypress's default network proxy port follows the system settings. However, my previous system's network proxy management was quite messy. Therefore, I restored my Windows computer's network settings to default, but Cypress's network proxy did not change. So, I ran` npx cypress clear cache` to clear the cache. After executing this command, we are required to run` npx cypress install `to restart the project. After restarting the project, Cypress's network proxy port was correct and it could access the network.
